Chore typedi

Sam09040 opened the pull request:
- pull request focused on adding typedi
- added typedi as lib;
- added service decorator;

Sam09040 committed:
implementing typedi

- added typedi as lib;
- added service decorator;

racsandoval commented on src/api/graphql/module/address/address.resolver.ts:
```diff
@@ -1,19 +1,23 @@
 export class AddressResolver {
-  constructor() {}
+  constructor(
+    private readonly addressUseCase = Container.get(AddressUseCase),
```
vale utilizar o typedi com os resolvers também, para não precisar desse `Container.get`

racsandoval commented on src/api/graphql/module/user/user.resolver.ts:
```diff
@@ -1,38 +1,45 @@
+import { CreateUserUseCase, LoginUseCase, UsersUseCase, UserUseCase } from '@domain/user';
 
+@Service()
 @Resolver()
 export class UserResolver {
-  constructor() {}
+  constructor(
+    private readonly userUseCase = Container.get(UserUseCase),
+    private readonly loginUseCase = Container.get(LoginUseCase),
+    private readonly usersUseCase = Container.get(UsersUseCase),
+    private readonly createUserUseCase = Container.get(CreateUserUseCase)
+  ) {}
```
esse está com o decorator de service mas tem os `Container.get` no constructor, não deveria precisar

Sam09040 committed:
fix: Container

- removed Container.get from constructor;
- added Container to buildSchema's container;

sophia-bergamo commented on src/api/graphql/module/user/user.resolver.ts:
```diff
@@ -1,38 +1,45 @@
-  user(@Arg('id', () => Int) id: number, @Ctx() context: ServerContext) {
-    return userUseCase(id, context.token);
+  async user(@Arg('id', () => Int) id: number, @Ctx() context: ServerContext) {
+    return await this.userUseCase?.exec(id, context.token);
```
```suggestion
    return this.userUseCase.exec(id, context.token);
```

sophia-bergamo commented on src/domain/user/create-user.use-case.ts:
```diff
@@ -4,53 +4,59 @@ import { JwtService } from '@core/security/jwt';
+    private readonly datasource: UserDbDataSource
+  ) {}
+
+  async exec (input: UserInputModel, token: string | undefined): Promise<UserModel> {
```
nas funções é sempre bom criar um único objeto de parâmetro pq evita problemas nas ordens quando essas funções forem chamadas

sophia-bergamo commented on src/domain/user/create-user.use-case.ts:
```diff
@@ -4,53 +4,59 @@ import { JwtService } from '@core/security/jwt';
+ 
+   const user = await this.datasource.insert(newUser);
+ 
+   return user;
```
```suggestion
   return this.datasource.insert(newUser);
```

sophia-bergamo commented on src/domain/user/users.use-case.ts:
```diff
@@ -2,29 +2,35 @@ import { UnauthorizedError } from '@core/error';
-      reason: 'A valid token must be provided.',
-    });
-  }
+  async exec (input: PageInputModel, token: string | undefined) {
```
mesma coisa aqui

Sam09040 committed:
small fix

- create use cases now returning just the response of the insert functions;
